x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -219,10 +218,17 @@ public void getCheckpointingInfo(DataFrameTransformsCheckpointService transforms
                 ));
     }
 
-    // Here `failOnConflict` is usually true, except when the initial start is called when the task is assigned to the node
-    synchronized void start(Long startingCheckpoint, boolean force, boolean failOnConflict, ActionListener<Response> listener) {
```

Comment:
Do I get this right? 

Due to this simplification we do not need #46347 anymore and revert it.

Reply:
pretty much. Since the ONLY thing that can call start against the task is the node executor, we know that it should be allowed to call start against a started task.